system/lib/wasmfs/thread_utils.h

```diff
@@ -22,39 +23,73 @@ namespace emscripten {
 // Helper class for synchronously proxying work to a dedicated worker thread,
 // including where the work is asynchronous.
 class ProxyWorker {
-  // The queue we use to proxy work and the dedicated worker.
+public:
+  enum Synchronicity { Sync, Async };
+
+private:
+  // The queue we use to proxy work.
   ProxyingQueue queue;
+
+  // Determines how we notify the caller that the worker is ready to receive
+  // work.
+  Synchronicity synchronicity;
```

Comment:
Why do you think about having two children of this class, `ProxyWorkerSync` and `Async`? Reading the code below I think maybe it might be easier to understand that way.

Reply:
Sure, I can play around with refactoring this.